extensions/ccui/base-classes/UIScale9Sprite.js

```diff
@@ -576,347 +604,305 @@ ccui.Scale9Sprite = cc.Node.extend(/** @lends ccui.Scale9Sprite# */{
             t = cc.affineTransformRotate(t, 1.57079633);
+        
+            leftTopBoundsOriginal = cc.rectApplyAffineTransform(leftTopBoundsOriginal, t);
+            centerBoundsOriginal = cc.rectApplyAffineTransform(centerBoundsOriginal, t);
+            rightBottomBoundsOriginal = cc.rectApplyAffineTransform(rightBottomBoundsOriginal, t);
+        
+            centerBounds = cc.rectApplyAffineTransform(centerBounds, t);
+            rightBottomBounds = cc.rectApplyAffineTransform(rightBottomBounds, t);
+            leftBottomBounds = cc.rectApplyAffineTransform(leftBottomBounds, t);
+            rightTopBounds = cc.rectApplyAffineTransform(rightTopBounds, t);
+            leftTopBounds = cc.rectApplyAffineTransform(leftTopBounds, t);
+            rightCenterBounds = cc.rectApplyAffineTransform(rightCenterBounds, t);
+            leftCenterBounds = cc.rectApplyAffineTransform(leftCenterBounds, t);
+            centerBottomBounds = cc.rectApplyAffineTransform(centerBottomBounds, t);
+            centerTopBounds = cc.rectApplyAffineTransform(centerTopBounds, t);
+        
+            rotatedLeftTopBoundsOriginal.x = leftTopBoundsOriginal.x;
+            rotatedCenterBoundsOriginal.x = centerBoundsOriginal.x;
+            rotatedRightBottomBoundsOriginal.x = rightBottomBoundsOriginal.x;
+        
+            rotatedCenterBounds.x = centerBounds.x;
+            rotatedRightBottomBounds.x = rightBottomBounds.x;
+            rotatedLeftBottomBounds.x = leftBottomBounds.x;
+            rotatedRightTopBounds.x = rightTopBounds.x;
+            rotatedLeftTopBounds.x = leftTopBounds.x;
+            rotatedRightCenterBounds.x = rightCenterBounds.x;
+            rotatedLeftCenterBounds.x = leftCenterBounds.x;
+            rotatedCenterBottomBounds.x = centerBottomBounds.x;
+            rotatedCenterTopBounds.x = centerTopBounds.x;
+        
+        
+            rotatedLeftTopBoundsOriginal.y = leftTopBoundsOriginal.y;
+            rotatedCenterBoundsOriginal.y = centerBoundsOriginal.y;
+            rotatedRightBottomBoundsOriginal.y = rightBottomBoundsOriginal.y;
+        
+            rotatedCenterBounds.y = centerBounds.y;
+            rotatedRightBottomBounds.y = rightBottomBounds.y;
+            rotatedLeftBottomBounds.y = leftBottomBounds.y;
+            rotatedRightTopBounds.y = rightTopBounds.y;
+            rotatedLeftTopBounds.y = leftTopBounds.y;
+            rotatedRightCenterBounds.y = rightCenterBounds.y;
+            rotatedLeftCenterBounds.y = leftCenterBounds.y;
+            rotatedCenterBottomBounds.y = centerBottomBounds.y;
+            rotatedCenterTopBounds.y = centerTopBounds.y;
+        }
+
+        // Centre
+        this._centre = new cc.Sprite();
+        this._centre.initWithTexture(selTexture, rotatedCenterBounds, rotated);
+        if(rotatedCenterBounds.width > 0 && rotatedCenterBounds.height > 0 )
+            this.addProtectedChild(this._centre);
+
+        // Top
+        this._top = new cc.Sprite();
```

Comment:
If the current `this._top` exist, you should use the existing object instead of creating a new one